lib/compat.php

```diff
@@ -9,6 +9,124 @@
  */
 
 if ( ! function_exists( 'register_block_type_from_metadata' ) ) {
+	/**
+	 * Removes the block asset's path prefix if provided.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $asset_handle_or_path Asset handle or prefixed path.
+	 *
+	 * @return string Path without the prefix or the original value.
+	 */
+	function gutenberg_remove_block_asset_path_prefix( $asset_handle_or_path ) {
+		$path_prefix = 'file://';
+		if ( strpos( $asset_handle_or_path, $path_prefix ) !== 0 ) {
+			return $asset_handle_or_path;
+		}
+		return substr(
+			$asset_handle_or_path,
+			strlen( $path_prefix )
+		);
+	}
+
+	/**
+	 * Generates the name for an asset based on the name of the block
+	 * and the field name provided.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $block_name Name of the block.
+	 * @param string $field_name Name of the metadata field.
+	 *
+	 * @return string Generated asset name for the block's field.
+	 */
+	function gutenberg_generate_block_asset_handle( $block_name, $field_name ) {
+		$field_mappings = array(
+			'editorScript' => 'editor-script',
+			'script'       => 'script',
+			'editorStyle'  => 'editor-style',
+			'style'        => 'style',
+		);
+		return str_replace( '/', '-', $block_name ) .
+			'-' . $field_mappings[ $field_name ];
+	}
+
+	/**
+	 * Finds a script handle for the selected block metadata field. It detects
+	 * when a path to file was provided and finds a corresponding
+	 * asset file with details necessary to register the script under
+	 * automatically generated handle name. It returns unprocessed script handle
+	 * otherwise.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param array  $metadata Block metadata.
+	 * @param string $field_name Field name to pick from metadata.
+	 *
+	 * @return string Script handle provided directly or created through script's registration.
+	 */
+	function gutenberg_get_script_handle( $metadata, $field_name ) {
```

Comment:
What would be the proper name to use in WordPress core? How about: `get_block_script_handle_from_asset`?

Reply:
Given that it calls `wp_register_script` I'd say `register_block_script_handle`

The `_from_asset` part seems redundant.

Reply:
Good hints, `register_block_script_handle` and `register_block_style_handle` it is 👍